inspire_disambiguation/core/ml/models.py

```diff
@@ -672,6 +676,24 @@ def fit(self, n_jobs=8):
         )
         self.clusterer.fit(self.X, self.y)
 
+    def score(self, validation_uuids, labels):
+        all_uuids = np.vectorize(lambda x: x.signature_uuid)(self.X).flatten().tolist()
+        validation_uuids_array = np.array(validation_uuids)
+        mask = np.array([item in validation_uuids_array for item in all_uuids])
+        y_tr = self.y[~mask]
+        y_val = self.y[mask]
+        labels_tr = self.labels_[~mask]
+        labels_val = labels[mask]
+        # concatenations to keep correct order
+        y = np.concatenate([y_tr, y_val])
+        labels = np.concatenate([y_tr, labels_tr])
+        uuids = np.concatenate([all_uuids[~mask], validation_uuids_array])
+        wrongly_classified_samples = uuids[y_val != labels_val]
```

Comment:
I don't think this works, as you might have a relabelling (look at how they do it for the B3 f-score computation).